test/index.test.js

```diff
@@ -55,51 +62,141 @@ describe('Parsely', function() {
+        analytics.stub(window.PARSELY, 'beacon');
+        analytics.stub(window.PARSELY.beacon, 'trackPageView');
+      });
+
+      it('should do nothing if dynamic tracking is not enabled', function() {
+        analytics.page();
+        analytics.didNotCall(window.PARSELY.beacon.trackPageView);
+      });
+
+      it('should call page if dynamic tracking is enabled', function() {
+        parsely.options.dynamicTracking = true;
+        analytics.page();
+        analytics.called(window.PARSELY.beacon.trackPageView);
+      });
+
+      it('should not pass metadata when not enabled', function() {
+        parsely.options.dynamicTracking = true;
+        analytics.page({
+          author: 'Chris Sperandio'
+        });
+        var args = window.PARSELY.beacon.trackPageView.args;
+        analytics.assert(!args[0][0].metadata);
+      });
+
+      it('should pass metadata when enabled', function() {
+        parsely.options.dynamicTracking = true;
+        parsely.options.inPixelMetadata = true;
+        analytics.page({
+          author: 'Chris Sperandio'
+        });
+        var args = window.PARSELY.beacon.trackPageView.args;
+        analytics.deepEqual(args[0][0].metadata, {
+          creator: 'Chris Sperandio',
+          url: document.location.href
+        });
+      });
+    });
+
+    describe('track', function() {
+      beforeEach(function() {
+        analytics.stub(window.PARSELY, 'beacon');
+        analytics.stub(window.PARSELY.beacon, 'trackPageView');
+      });
+
+      it('should do nothing if events are not enabled', function() {
+        analytics.track('test');
+        analytics.didNotCall(window.PARSELY.beacon.trackPageView);
+      });
+
+      it('should send events if enabled', function() {
+        parsely.options.trackEvents = true;
+        analytics.track('test');
+        analytics.called(window.PARSELY.beacon.trackPageView);
+      });
+
+      it('should send event name as `action`', function() {
+        parsely.options.trackEvents = true;
+        analytics.track('test');
+        var args = window.PARSELY.beacon.trackPageView.args;
+        analytics.assert(args[0][0].action === 'test');
+      });
+
+      it('should send event properties as `data`', function() {
+        parsely.options.trackEvents = true;
+        analytics.track('test', { testing: 'test' });
+        var args = window.PARSELY.beacon.trackPageView.args;
+        analytics.deepEqual(args[0][0].data, { testing: 'test' });
+      });
+    });
   });
 });
 
 function isParselyMetaTag(element) {
   return !!(element && element.getAttribute('data-parsely-site'));
 }
+
+function isPjsScript(element) {
+  return !!element && (/p.js$/).test(element.src);
+}
+
+function isLoaded() {
+  return !!filter(document.getElementsByTagName('script'), isPjsScript).length;
+}
+
+/**
+ * Fully Clear Cookies.
+ *
+ * Sandbox only removes the values, and Parsely's script errors
+ * on empty values for their cookie keys
+ */
+
+function clearCookies() {
```

Comment:
once we release https://github.com/segmentio/clear-cookies/pull/2 we can remove the inline implementation